Kernel/Devices/VMWareSVGADevice.h

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2021, the Serenity OS Developers
+ * Copyright (c) 2021, Alexander Richards <electrodeyt@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <AK/Vector.h>
+#include <Kernel/Devices/BlockDevice.h>
+#include <Kernel/IO.h>
+#include <Kernel/Interrupts/IRQHandler.h>
+#include <Kernel/PCI/Device.h>
+
+namespace Kernel {
+class VMWareSVGADevice final : public BlockDevice
+    , public IRQHandler {
+public:
+    static void initialize();
+    static VMWareSVGADevice& the();
+
+    VMWareSVGADevice();
+
+    virtual int ioctl(FileDescription&, unsigned request, FlatPtr arg) override;
+    virtual KResultOr<Region*> mmap(Process&, FileDescription&, const Range&, u64 offset, int prot, bool shared) override;
+
+    virtual mode_t required_mode() const override { return 0660; }
+    virtual String device_name() const override;
+
+private:
+    virtual const char* class_name() const override { return "VMWareSVGA"; }
+    virtual bool can_read(const FileDescription&, size_t) const override { return true; }
+    virtual bool can_write(const FileDescription&, size_t) const override { return true; }
+    virtual void start_request(AsyncBlockDeviceRequest& request) override { request.complete(AsyncDeviceRequest::Failure); }
+    virtual KResultOr<size_t> read(FileDescription&, u64, UserOrKernelBuffer&, size_t) override { return -EINVAL; }
+    virtual KResultOr<size_t> write(FileDescription&, u64, const UserOrKernelBuffer&, size_t) override { return -EINVAL; }
+
+    virtual void handle_irq(const RegisterState&) override;
+
+    enum class IORegister {
+        ID = 0,
+        Enable,
+        Width,
+        Height,
+        MaxWidth,
+        MaxHeight,
+        Depth,
+        BitsPerPixel,
+        PseudoColor,
+        RedMask,
+        GreenMask,
+        BlueMask,
+        BytesPerLine,
+        FBStart,
+        FBOffset,
+        VRamSize,
+        FBSize,
+        Capabilities,
+        MemStart,
+        MemSize,
+        ConfigDone,
+        Sync,
+        Busy,
+        GuestID,
+        CursorID,
+        CursorX,
+        CursorY,
+        CursorOn,
+        HostBitsPerPixel,
+        ScratchSize,
+        MemRegs,
+        NumDisplays,
+        Pitchlock,
+        IRQMask,
+        NumGuestDisplays,
+        DisplayID,
+        DisplayIsPrimary,
+        DisplayPositionX,
+        DisplayPositionY,
+        DisplayWidth,
+        DisplayHeight,
+        GMRID,
+        GMRDescriptor,
+        GMRMaxIDS,
+        GMRMaxDescriptorLength,
+        Traces
+    };
+
+    enum class FifoRegister {
+        Min = 0,
+        Max,
+        NextCMD,
+        Stop,
+        Capabilites = 4,
```

Comment:
```suggestion
        Capabilities,
```
The "4" is already set by the incremental values, so it seems weird to say it just for this one.

Reply:
Yeah, i basically looked at the offical header, went "wtf? why did they do this?" and just added it while i was trying to get it to work.